napari/_vispy/vispy_base_layer.py

```diff
@@ -93,6 +94,9 @@ def scale(self):
 
     @scale.setter
     def scale(self, scale):
+        # Avoid useless update if nothing changed in the displayed dims:
+        if self.scale is not None and np.all(self.scale[:2] == scale[:2]):
```

Comment:
I'm confused about why this is `:2` instead of `-2:`? And what about 3D?

Reply:
(ie, it would be good to add a comment here and on `translate` about what dimensions exactly this is comparing.)

Reply:
it's because the vispy properties are always 4-vectors, i added a comment and padded the scale, translate so it works in all 2D and 3D cases too, good catch